libtock/tock.c

```diff
@@ -102,6 +103,24 @@ void yield_for(bool *cond) {
   }
 }
 
+int quick_yield(void* base) {
+  if (task_cur != task_last) {
+    tock_task_t task = task_queue[task_cur];
+    task_cur = (task_cur + 1) % TASK_QUEUE_SIZE;
+    task.cb(task.arg0, task.arg1, task.arg2, task.ud);
+    return 1;
+  } else {
+    uint32_t tasks = ros_get_pending_tasks(base);
+
+    if (tasks > 0) {
+      // Waiting tasks, call yield
+      yield();
+    }
+
+    return tasks;
+  }
+}
```

Comment:
It seems like this should be implemented in `ros.c`, so this file does not have a dependency on `ros.h`, and so that it is clear that this operation is only possible on boards with the `ros` capsule installed.